### PR TITLE
Ensure backwards compatibility with Prebid Mobile v0.0.1 for User field

### PR DESCRIPTION
**Pull request by nhedley**
In Prebid Mobile v0.0.1 on both iOS and Android user gender is passed as an integer. We recently found out that clients are using v0.0.1 and need to ensure backwards compatibility with that version.

This PR makes the "user" param of PBSRequest a rawJSONParameter so gender can be passed through as a string or integer without throwing an error. If we are not using v0.0.1 of the SDK, we then parse the User object into the openrtb.User object. 

I'm not changing the static pbs_request.json here because we still want the user object to be passed in according to OpenRTB standards.

### CODE REVIEW

File: pbs/pbsrequest.go [hunk 1]
```diff
@@ -127,6 +135,17 @@ func ParsePBSRequest(r *http.Request, cache cache.Cache) (*PBSRequest, error) {
 	if pbsReq.Device == nil {
 		pbsReq.Device = &openrtb.Device{}
 	}
+	if pbsReq.SDK == nil {
+		pbsReq.SDK = &SDK{}
+	}
+	if pbsReq.SDK.Version != "0.0.1" {
+		if pbsReq.PBSUser != nil {
```
Comment (anwzhang) [inline pbs/pbsrequest.go]: What happens if SDK version == 0.0.1? Should it change the gender to string and then forward to bidders?

Comment (nhedley) [inline pbs/pbsrequest.go]: I don't think its necessary to do that at the moment. We currently don't forward the user object to all bidders in Prebid Server, so if we want to add that capability we should do it in a separate PR.

File: pbs/pbsrequest.go [hunk 2]
```diff
@@ -81,10 +87,12 @@ type PBSRequest struct {
 	IsDebug       bool            `json:"is_debug"`
 	App           *openrtb.App    `json:"app"`
 	Device        *openrtb.Device `json:"device"`
-	User          *openrtb.User   `json:"user"`
+	PBSUser       json.RawMessage `json:"user"`
+	SDK           *SDK            `json:"sdk"`
```
Comment (mkendall07) [inline pbs/pbsrequest.go]: is SDK supported on our `/openRTB2` endpoint?  Do we expect to send it in the future, or is it just for checking the version?

File: pbs/pbsrequest.go (hunk 1 above)
Comment (mkendall07) [inline pbs/pbsrequest.go]: My concern with this is that it's specific to the AN adapter, where the PBS request is applicable to all adapters. Is there a way to move this logic into AN adapter?

Comment (nhedley) [inline pbs/pbsrequest.go]: The code isn't specific to the AN adapter actually. The issue here is that we used to pass the User gender field from Prebid Mobile (in v0.0.1) as an integer, and for Facebook demand they require the User object, where the gender field in OpenRTB is a string. When i implemented adding the User object here for Facebook demand (https://github.com/prebid/prebid-server/pull/94), I also updated Prebid Mobile to pass a string and we released that version as 0.0.2.

However, we recently realized we needed to also support the older version of Prebid Mobile v0.0.1 where we were passing an integer. This PR's purpose is to allow User gender to be passed as either a string or an integer to Prebid Server without getting a parsing error. Very open to a better way to solve this problem.

File: pbs/pbsrequest.go (hunk 2 above)
Comment (nhedley) [inline pbs/pbsrequest.go]: `sdk` is a field specific to Prebid Mobile and is not an OpenRTB field. I didn't add it to the static pbs_request.json for that reason. Its for internal use to pass Android/iOS and version in case we need to support certain behaviors, like this backwards compatibility case.

File: pbs/pbsrequest.go (hunk 1 above)
Comment (mkendall07) [inline pbs/pbsrequest.go]: ok that makes sense then.

**Comment by dbemiller** [comment]
I chatted with @nhedley a bit offline. She pointed out that `openrtb_util` is populating the `User` field on the request using cookie data.

I'm concerned by all the "app vs. non-app" logic spread throughout the PBS code. Ideally we should work towards a state where (to the greatest extent possible) that logic is kept as close to the API as possible.

I'm merging this because they're time crunched... but in a later PR, she said she'd consolidate the "non-app" User logic from `openrtb_util.makeOpenRTBGeneric` alongside the logic here, to keep it all in one place.
